crates/ruff_linter/src/rules/ruff/snapshots/ruff_linter__rules__ruff__tests__RUF012_RUF012.py.snap

```diff
@@ -40,3 +40,69 @@ RUF012.py:89:38: RUF012 Mutable class attributes should be annotated with `typin
 90 |
 91 | from sqlmodel import SQLModel
    |
+
+RUF012.py:114:36: RUF012 Mutable class attributes should be annotated with `typing.ClassVar`
+    |
+112 |     }
+113 |
+114 |     mutable_default: 'list[int]' = []
+    |                                    ^^ RUF012
+115 |     immutable_annotation: 'Sequence[int]'= []
+116 |     without_annotation = []
+    |
+
+RUF012.py:115:44: RUF012 Mutable class attributes should be annotated with `typing.ClassVar`
+    |
+114 |     mutable_default: 'list[int]' = []
+115 |     immutable_annotation: 'Sequence[int]'= []
+    |                                            ^^ RUF012
+116 |     without_annotation = []
+117 |     class_variable: 'ClassVar[list[int]]' = []
+    |
+
+RUF012.py:116:26: RUF012 Mutable class attributes should be annotated with `typing.ClassVar`
+    |
+114 |     mutable_default: 'list[int]' = []
+115 |     immutable_annotation: 'Sequence[int]'= []
+116 |     without_annotation = []
+    |                          ^^ RUF012
+117 |     class_variable: 'ClassVar[list[int]]' = []
+118 |     final_variable: 'Final[list[int]]' = []
+    |
+
+RUF012.py:117:45: RUF012 Mutable class attributes should be annotated with `typing.ClassVar`
+    |
+115 |     immutable_annotation: 'Sequence[int]'= []
+116 |     without_annotation = []
+117 |     class_variable: 'ClassVar[list[int]]' = []
```

Comment:
Hmm, why are we emitting an error here? It looks like it already is annotated with ClassVar?

Reply:
Ahh, and I see this isn't a regression; we also have this false negative on `main`. Feel free to merge, in that case!

Probably a lot more rules should be using this helper function when they inspect annotations: https://github.com/astral-sh/ruff/blob/2cf6822537b2c126bdda5232bba195289531c559/crates/ruff_linter/src/checkers/ast/mod.rs#L453-L471

(I introduced the helper in #12951 ;)